resources/qml/UserSettingsPage.qml

```diff
@@ -0,0 +1,218 @@
+
+        palette: Nheko.colors
+        ScrollBar.horizontal.visible: false
+        anchors.fill: parent
+        anchors.margins: Nheko.paddingLarge
+
+        contentWidth: availableWidth
+
+        Timer {
+            id: deadTimer
+            interval: 500
+        }
+
+        Connections {
+            target: scroll.contentItem
+            function onContentYChanged() { deadTimer.restart(); }
+        }
+
+
+        GridLayout {
+            id: grid
+
+            columns: userSettingsDialog.collapsed ? 1 : 2
+            rowSpacing: Nheko.paddingMedium
+            columnSpacing: Nheko.paddingMedium
+
+            anchors.fill: parent
+            anchors.leftMargin: userSettingsDialog.collapsed ? Nheko.paddingLarge : (userSettingsDialog.width-600) * 0.4
+            anchors.rightMargin: userSettingsDialog.collapsed ? Nheko.paddingLarge : (userSettingsDialog.width-600) * 0.4
+
+            Repeater {
+                model: UserSettingsModel
+
+                delegate: Item {
+                    required property var model
+                    id: r
+
+                    Component.onCompleted: {
+                        while (children.length) { 
+                            console.log("Reparenting: " + children[0]);
+                            children[0].parent = grid;
+                        }
+                    }
+
+                    Label {
+                        Layout.alignment: Qt.AlignLeft
+                        Layout.fillWidth: true
+                        color: Nheko.colors.text
+                        text: model.name
+                        //Layout.column: 0
+                        Layout.columnSpan: (model.type == UserSettingsModel.SectionTitle && !userSettingsDialog.collapsed) ? 2 : 1
+                        //Layout.row: model.index
+                        Layout.minimumWidth: implicitWidth
+                        Layout.leftMargin: model.type == UserSettingsModel.SectionTitle ? 0 : Nheko.paddingMedium
+                        Layout.topMargin: model.type == UserSettingsModel.SectionTitle ? Nheko.paddingLarge : 0
+                        font.pointSize: 1.1 * fontInfo.pointSize
+
+                        HoverHandler {
+                            id: hovered
+                            enabled: model.description ?? false
+                        }
+                        ToolTip.visible: hovered.hovered && model.description
+                        ToolTip.text: model.description ?? ""
+                        ToolTip.delay: Nheko.tooltipDelay
+                    }
+
+                    DelegateChooser {
+                        id: chooser
+
+                        roleValue: model.type
+                        Layout.alignment: Qt.AlignRight
+
+                        //Layout.column: model.type == UserSettingsModel.SectionTitle ? 0 : 1
+                        Layout.columnSpan: (model.type == UserSettingsModel.SectionTitle && !userSettingsDialog.collapsed) ? 2 : 1
```

Comment:
This logic probably isn't needed for these elements.  I don't think we have a case where the Section Title also has a toggle/dropdown associated with it.

Reply:
It is. The Title and the line below it are different Items and that makes the handling much easier.